### DISCUSSION TITLE
Question about input stream bitrate

### DISCUSSION THREAD

**Discussion by jbrepogmailcom** [librespot-org/librespot#904] [Feature Requests]
Hello! I understand that setting 320kbps is setting for output quality, which is then streamed as PCM to audio device - that is why all songs from spotify show as PCM 1440kbps (in KODI).

I am wondering if it is also possible to set quality of input stream that is requested from spotify servers and how to see that quality.

Thank you

**Comment by roderickvd** [comment]
I don't know about Kodi, but you could use the passthrough decoder and pass Ogg directly with `--passthrough`.

**Comment by JasonLG1979** [comment]
The bitrate option in librespot is the Spotify stream quality as in 320 kbps ogg vorbis for example. What you are seeing is the PCM bitrate as in the bitrate of 16 bit 44.1 stereo PCM.

I have no idea how to configure Kodi. You would need ask them?
